lib/active_model/serializer/adapter/json_api/deserialization.rb

```diff
@@ -0,0 +1,73 @@
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        module Deserialization
+          module_function
+
+          # Parse a Hash or ActionController::Parameters representing a JSON API document
+          # into an ActiveRecord-ready hash.
+          #
+          # @param [Hash|ActionController::Parameters] document
+          # @param [Hash] options
+          #   fields: Array of symbols and a Hash. Specify whitelisted fields, optionally
+          #      specifying the attribute name on the model.
+          # @return [Hash] ActiveRecord-ready hash
+          #
+          def parse(document, options = {})
```

Comment:
The issue with `from_json` is that it should be called on a model. Here, we make no assumption about the model. I guess we could override `ActiveModel::Serializers::JSON` so that it calls the `parse` method and updates the model. But it would be JSON API specific. Thoughts?